pkg/utils/k8sutil.go

```diff
@@ -265,247 +268,98 @@ func GetFunctionData(runtime, ftype, modName string) (imageName, depName, fileNa
 	return
 }
 
+type metadata struct {
```

Comment:
This should not exist - see below.  If you create a "poor-man's Function" like this, then you're just going to spend your whole time converting to/from the "real" Function and passing extra parameters everywhere to make up for all the lost information.